src/sql/workbench/contrib/queryplan2/browser/queryPlan.ts

```diff
@@ -0,0 +1,738 @@
+		// 	let diagramNode: any = diagram_stack.pop();
+		// 	if (node) {
+		// 		diagramNode.label = node.name;
+		// 		if (node.edges) {
+		// 			for (let i = 0; i < node.edges.length; ++i) {
+		// 			}
+		// 		}
+		// 	}
+		// }
+
+		diagramNode.label = node.name;
+		if (node.properties && node.properties.length > 0) {
+			diagramNode.metrics = node.properties.map(e => { return { name: e.name, value: e.formattedValue.substring(0, 75) }; });
+		}
+
+		let icon = this._iconMap[node.type];
+		if (icon) {
+			diagramNode.icon = icon;
+		}
+
+		if (node.children) {
+			diagramNode.children = [];
+			for (let i = 0; i < node.children.length; ++i) {
+				diagramNode.children.push(this.populate(node.children[i], new Object()));
+			}
+		}
+		return diagramNode;
+	}
+
+	private createPlanDiagram(container: HTMLDivElement): void {
+		let diagramRoot: any = new Object();
+		let graphRoot: azdata.QueryPlanGraphNode = this._graph.root;
+		this.populate(graphRoot, diagramRoot);
+
+		new azdataGraph.azdataQueryPlan(container, diagramRoot, this._iconPaths);
+	}
+
+
+	public set graph(graph: azdata.QueryPlanGraph | undefined) {
+		this._graph = graph;
+		if (this._graph) {
+			/**
+			 * Create a show plan graph here.
+			 */
+			this._container.appendChild(document.createTextNode(`Query ${this._graphIndex}: `));
+			this._relativeCost = document.createTextNode('(relative to the script):');
+			this._container.appendChild(this._relativeCost);
+			this._container.appendChild(document.createElement('br'));
+			this._container.appendChild(document.createTextNode(`${graph.query}`));
+			let diagramContainer = document.createElement('div');
+			this.createPlanDiagram(diagramContainer);
+			this._container.appendChild(diagramContainer);
+
+			this._container.appendChild(document.createTextNode('Need to add graph control here'));
+			this._container.appendChild(document.createElement('br'));
+			this._container.appendChild(this.propContainer);
+			this.setData(this._graph.root.properties);
+			this._container.appendChild(this._actionBarContainer);
+		}
+	}
+
+	public get graph(): azdata.QueryPlanGraph | undefined {
+		return this._graph;
+	}
+
+	public set relativeCost(newCost: number) {
+		this._relativeCost.nodeValue = `(relative to the script): ${newCost.toFixed(2)}%`;
+	}
+
+	public setData(props: azdata.QueryPlanGraphElementProperty[]): void {
+		this._data = [];
+		props.forEach((p, i) => {
+			this._data.push({
+				id: p.name,
+				name: p.name,
+				propValue: p.formattedValue,
+				_collapsed: true
+			});
+		});
+		this._dataView.beginUpdate();
+		this._dataView.setItems(this._data);
+		this._dataView.endUpdate();
+		this._dataView.refresh();
+		this._table.autosizeColumns();
+		this._table.updateRowCount();
+		this._table.resizeCanvas();
+		this._table.render();
+	}
+}
+
+/**
+ * Registering a feature flag for query plan.
+ * TODO: This should be removed before taking the feature to public preview.
+ */
+const QUERYPLAN2_CONFIG_ID = 'queryPlan2';
```

Comment:
@Charles-Gagnon, here is the feature flag used for enabling/disabling this feature.